riscv/csr_init.cc

```diff
@@ -205,7 +205,11 @@ void state_t::csr_init(processor_t* const proc, reg_t max_isa)
     add_csr(CSR_TDATA2, tdata2 = std::make_shared<tdata2_csr_t>(proc, CSR_TDATA2));
     add_csr(CSR_TDATA3, std::make_shared<tdata3_csr_t>(proc, CSR_TDATA3));
     add_csr(CSR_TINFO, std::make_shared<tinfo_csr_t>(proc, CSR_TINFO));
-    add_csr(CSR_TCONTROL, tcontrol = std::make_shared<masked_csr_t>(proc, CSR_TCONTROL, CSR_TCONTROL_MPTE | CSR_TCONTROL_MTE, 0));
+    if (proc->extension_enabled_const('S')) {
+      add_csr(CSR_TCONTROL, tcontrol = std::make_shared<masked_csr_t>(proc, CSR_TCONTROL, 0, 0));
+    } else {
+      add_csr(CSR_TCONTROL, tcontrol = std::make_shared<masked_csr_t>(proc, CSR_TCONTROL, CSR_TCONTROL_MPTE | CSR_TCONTROL_MTE, 0));
+    }
```

Comment:
This only partially consists of my reading on the spec

The spec says that triggers with action=0 do not match/fire in M-mode if ```tcontrol.mte=0```. Although the spec says that only Solution 2 implements ```tcontrol.mte```, I don't feel software doing Solution 1 expects ```tcontrol.mte``` to be read-only 0. (In such a case, I would be confused if the trigger fires with a read-only ```tcontrol.mte=0```.)

Nevertheless, I agree that the behavior does not violate the wording of the spec and would like to let others decide.